### PR TITLE
Sorted set blocking commands

### PR DESCRIPTION
**Pull request by gheorghitamutu**
Hello,

The feature has been requested in #944 & #476.

The commands implemented are:
https://redis.io/commands/bzpopmin
https://redis.io/commands/bzpopmax
https://redis.io/commands/bzmpop

Tests were added for all 3 of them.

### CODE REVIEW

**Comment by gheorghitamutu** [comment]
As a side note:
The `BZMPOP` command is available from version 7.0.0 and it will 
probably fail testing with Redis version 6.2.4. If there's a workaround/check that I can do in order to specify Redis' version, please let me know. If not, I'll remove this specific command test.

File: redis/tests/test_basic.rs [hunk 1]
```diff
@@ -1280,3 +1280,32 @@ fn test_set_options_options() {
...
+    assert_eq!(con.zadd("a", "1a", 1), Ok(()));
+    assert_eq!(con.zadd("b", "2b", 2), Ok(()));
+    assert_eq!(con.zadd("c", "3c", 3), Ok(()));
+    assert_eq!(con.zadd("d", "4d", 4), Ok(()));
+    assert_eq!(con.zadd("a", "5a", 1), Ok(()));
+    assert_eq!(con.zadd("b", "6b", 2), Ok(()));
+    assert_eq!(con.zadd("c", "7c", 3), Ok(()));
+    assert_eq!(con.zadd("d", "8d", 4), Ok(()));
+
+    let result_multiple_min = con
```
Comment (jaymell) [inline redis/tests/test_basic.rs:1298]: Did you mean to only test the `min` command variants?

**Comment by jaymell** [comment]
RE the version checking, I'm surprised, but not aware of any place where we've previously excluded tests based on version. It should be possible to parse the server version out of `INFO` command and exclude the the specific assertions as needed, so I'd prefer to do that if you think you can tackle it.

**Comment by nihohit** [comment]
> It should be possible to parse the server version out of INFO command and exclude the the specific assertions as needed

maybe add a helper function that takes a version and returns bool to tests/utilities/mod.rs to do that, and return early from tests when the version is too early?
alternatively, maybe we can get the version from env variable?
 If you need help, please say.

File: redis/tests/test_basic.rs (hunk 1 above)
Comment (gheorghitamutu) [reply redis/tests/test_basic.rs:1298]: You're right, the max variables should have been the result of max functions. My mistake. 
I fixed it in [a08aba1](https://github.com/redis-rs/redis-rs/pull/962/commits/a08aba18f1b2ac8ebf50e7328a271761f0048c6a).

**Comment by gheorghitamutu** [comment]
@jaymell I did not see any tests for other (non) blocking functions added as commands that were introduced in Redis version 7.0.0 so that might be why there was no need for a version checking helper function.

@jaymell @nihohit I can probably do both of them as helpers for version checking but if you want a specific one, please let me know. Unfortunately, I'm not aware of a standard env variable that specifies the Redis version. If there's no such thing, I'd go with @jaymell suggestion and parse the INFO command for version information. If there's a custom env var set up, anyone who tries to create a build & test workflow needs to make sure that they also set it up and it sounds like another small dependency added.

**Comment by nihohit** [comment]
you're right, best to test the server version.

**Comment by gheorghitamutu** [comment]
@jaymell @nihohit please let me know what you think about statements below.

I've looked a bit into how I'd be doing this for just my case.
The quick and dirty code block I came with is something along these lines:
```rs
    let major_version : u32 = {
        let raw_info: String = redis::cmd("INFO").query(&mut con).unwrap();
        let pattern = "redis_version:";
        let version_position = raw_info.find(pattern).unwrap();
        let major_version = raw_info.chars().nth(version_position + pattern.len()).unwrap();
        major_version.to_digit(10).unwrap()
    };
```

What this solves is solely my current issue (partially, as non multi commands - `BZPOPMIN` & `BZPOPMAX` are only available from Redis version 5 as well).
What it doesn't is along these lines:
- there will probably be more tests in the future depending on the Redis version and thus this is not the right place nor the ergonomic implementation for this function/code block
- the function should probably be a helper for TestContext (and maybe keep it as metadata/ structure member) and this is out of the scope of the current PR (it should also take in account the actual format of the version `<major>.<minor>.<path>` in case there are cases we also want to look into that)
- the future design should also probably somehow remove the platform attribute (windows) for several tests and rely on the actual Redis version
- I'd rather open an entirely new issue based on the above

If you agree I can either merge the above code block in my test only or I can remove asserting/calling `BZMPOP` command that requires Redis version 7.

**Comment by nihohit** [comment]
@gheorghitamutu please try rebasing over https://github.com/redis-rs/redis-rs/pull/963 and using the functions I added there.

**Comment by gheorghitamutu** [comment]
Hello, @nihohit, thank you!

**Comment by nihohit** [comment]
@jaymell LGTM

**Comment by gheorghitamutu** [comment]
There was a small fix to be done in [8f26d05](https://github.com/redis-rs/redis-rs/pull/962/commits/8f26d056309dcdacad2ff95bedd9fa0efede4a39).

**Review by nihohit — APPROVED** [review]

**Comment by jaymell** [comment]
Thanks for the PR!

**Pull request merged**